packages/plugin-react/src/index.ts

```diff
@@ -122,13 +122,27 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
   const viteBabel: Plugin = {
     name: 'vite:react-babel',
     enforce: 'pre',
-    config() {
+    config(_, { mode }) {
+      const isProduction =
```

Comment:
This is simply copied from https://github.com/vitejs/vite/blob/4e9bdd4fb3654a9d43917e1cb682d3d2bad25115/packages/vite/src/node/config.ts#L488-L490

As far as I could tell, there doesn't seem to be a way to access `config.isProduction` within the `config()` hook. Obviously it would be preferable to have a single source of truth for this, but for simplicity I duplicated the logic to infer from environment and mode.

Maybe this should be exposed to `config()` directly or maybe `isProduction` should be extracted into a helper function that takes the environment and mode.